src/auth/auth.guard.ts

```
@@ -0,0 +1,46 @@
+
+    const request = context.switchToHttp().getRequest<Request>();
+    const authHeader = request.headers.authorization;
+    if (!authHeader) {
+      throw new UnauthorizedException('Authorization header missing');
+    }
+    const token = authHeader.split(' ')[1];
+    try {
+      //Step 1: Correctly Generate Encryption Key
+      const secret_key = process.env.JOSE_SECRET || '';
+      const hash = createHash('sha256').update(secret_key).digest();
+  
```

ESLint

[error] Delete `··`

(prettier/prettier)